parsers/IL.py

```diff
@@ -44,15 +60,19 @@ def fetch_price(
 ) -> dict:
     """Fetch price from Noga API."""
     if target_datetime is not None:
+        logger.warning(
+            "IL parser is not yet able to parse past dates, ignoring target_datetime"
+        )
         raise NotImplementedError("This parser is not yet able to parse past dates")
-
-    with get(IEC_PRICE) as response:
-        data = response.json()
+    if session is None:
+        session = Session()
+    response = session.get(IEC_PRICE)
+    data = response.json()
```

Comment:
```suggestion
    response: Response = session.get(IEC_PRICE)
    if not response.ok:
        raise ParserException("IL.py", f"The returned response was not okay and returned the status: {response.status_code}", zone_key)
    data = response.json()
```

Here is an example on how to use ParserException as well.